gtfs/spec/en/reference.md

```diff
@@ -388,26 +390,26 @@ In regions that have multiple official languages, transit agencies/operators typ
-|  `record_sub_id` | ID | **Conditionally Required** | Helps the record that contains the field to be translated when the table doesn’t have a unique ID. Therefore, the value in `record_sub_id` is the secondary ID of the table, as defined by the table below:<br><br>- None for `agency.txt`;<br>- None for `stops.txt`;<br>- None for `routes.txt`;<br>- None for `trips.txt`;<br>- `stop_sequence` for `stop_times.txt`;<br>- None for `pathways.txt`;<br>- None for `levels.txt`;<br>- None for `attributions.txt`.<br><br>No field should be translated in the other tables. However producers sometimes add extra fields that are outside the official specification and these unofficial fields may need to be translated. Below is the recommended way to use `record_sub_id` for those tables:<br><br>- None for `calendar.txt`;<br>- `date` for `calendar_dates.txt`;<br>- None for `fare_attributes.txt`;<br>- `route_id` for `fare_rules.txt`;<br>- None for `shapes.txt`;<br>- `start_time` for `frequencies.txt`;<br>- `to_stop_id` for `transfers.txt`.<br><br>Conditionally Required:<br>- **Forbidden** if `table_name` is `feed_info`.<br>- **Forbidden** if `field_value` is defined.<br>- **Required** if `table_name=stop_times` and `record_id` is defined. |
+|  `record_id` | ID | **Conditionally Required** | Defines the record that corresponds to the field to be translated. The value in `record_id` should be a main ID of the table, as defined below:<br><br>- `agency_id` for `agency.txt`<br>- `stop_id` for `stops.txt`;<br>- `route_id` for `routes.txt`;<br>- `trip_id` for `trips.txt`;<br>- `trip_id` for `stop_times.txt`;<br>- `pathway_id` for `pathways.txt`;<br>- `level_id` for `levels.txt`;<br>- `attribution_id` for `attribution.txt`.<br><br>Fields in tables not defined above should not translated. However producers sometimes add extra fields that are outside the official specification and these unofficial fields may be translated. Below is the recommended way to use `record_id` for those tables:<br><br>- `service_id` for `calendar.txt`;<br>- `service_id` for `calendar_dates.txt`;<br>- `fare_id` for `fare_attributes.txt`;<br>- `fare_id` for `fare_rules.txt`;<br>- `shape_id` for `shapes.txt`;<br>- `trip_id` for `frequencies.txt`;<br>- `from_stop_id` for `transfers.txt`.<br><br>Conditionally Required:<br>- **Forbidden** if `table_name` is `feed_info`.<br>- **Forbidden** if `field_value` is defined.<br>- **Required** if `field_value` is empty. |
```

Comment:
Should this be as follows:

"The value in `record_id` should be a main ID of the table, as defined" -> "The value in `record_id` must be a main ID of the table, as defined..."

Reply:
I think making the first 8 `id` fields for the official tables defined as requirements ("must") is good. 

How would the recommended `id` fields for unofficial tables be handled in a validator? A warning in all cases that an "unofficial" `id` is present and an error if none of the `id`s described in this description are present?

Reply:
@barbeau please correct me if I'm wrong. As of now the gtfs-validator does not support yet the validation of conditionally required fields. 
However, this could be handled as follows:
- if `table_name` is feed_info, and `record_id` is defined: that is an ERROR;
- if `field_value` is defined,  and `record_id` is defined: that is an ERROR;
- if `field_value` is defined, and `record_id` is not defined: that is an ERROR.
⬆️ these would apply to both official and unofficial tables. 

Now, when it comes to a situation when an ERROR from the previous step was not generated - in the case of an unofficial table:

>  A warning in all cases that an "unofficial" id is present

Yes - but I'd like to discuss this further during our next technical meeting. That encompasses a broader feature to be added: conditional requirements.

Reply:
I committed a change to make the official files "must" as described above.